File: EventPlanner.cpp
#include "EventPlanner.hpp"
#include "TrainingSession.hpp"
#include <stdexcept>

// Конструктор: Инициализирует поля
EventPlanner::EventPlanner(const std::list<std::string>& plannedEvents, const std::string& notifyRules)
    : plannedEvents(plannedEvents), notifyRules(notifyRules) {}

// Конструктор копирования
EventPlanner::EventPlanner(const EventPlanner& other)
    : plannedEvents(other.plannedEvents), notifyRules(other.notifyRules) {}

// Деструктор
EventPlanner::~EventPlanner() {}

// Метод: Добавляет в расписание (заглушка)
std::string EventPlanner::addToSchedule(const TrainingSession& session) {
    try {
        return "Событие добавлено в расписание"; // Пример
    }
    catch (const std::exception& e) {
        throw std::runtime_error("Ошибка при добавлении в расписание: " + std::string(e.what()));
    }
}

// Метод: Откладывает уведомление (заглушка)
void EventPlanner::delayNotification(const std::string& eventId, int delayMin) {
    try {
        if (eventId.empty()) {
            throw std::invalid_argument("ID события не может быть пустым");
        }
        if (delayMin < 0) {
            throw std::invalid_argument("Задержка не может быть отрицательной");
        }
        // Отложить
    }
    catch (const std::exception& e) {
        throw;
    }
}

// Метод: Переносит пропущенные занятия (заглушка)
void EventPlanner::rescheduleMissed() {
    // Перенос
}

File: FitnessAppCsharp/EventPlanner.cs
using System;
using System.Collections.Generic;

public class EventPlanner
{
    // Поля класса
    private List<string> plannedEvents;     // Список запланированных событий (например, "2025-04-05 18:00 - Тренировка ног")
    private string notifyRules;             // Правила уведомлений (например, "за 30 минут", "утром в 9:00")

    // Конструктор: инициализирует календарь и правила напоминаний
    public EventPlanner(List<string> plannedEvents, string notifyRules)
    {
        this.plannedEvents = plannedEvents;
        this.notifyRules = notifyRules;
    }

    // Метод: добавляет тренировку в расписание (возвращает статус)
    public string AddToSchedule(TrainingSession session)
    {
        
        plannedEvents.Add($"Тренировка: {session.GetSessionName()} - {DateTime.Now:yyyy-MM-dd HH:mm}");
        return "Scheduled"; // как в C++ версии
    }

    // Метод: откладывает уведомление для события
    public void DelayNotification(string eventId, int delayMin)
    {
        // Заглушка 
        Console.WriteLine($"Уведомление для {eventId} отложено на {delayMin} минут");
    }

    // Метод: переносит пропущенные тренировки на следующий доступный день
    public void RescheduleMissed()
    {
        // Заглушка
        Console.WriteLine("Пропущенные тренировки перенесены автоматически");
    }
}